src/utils/general.utils.ts

```diff
@@ -231,8 +231,57 @@ export function formatAmountWithDecimals({ amount, decimals }: { amount: number;
     return formattedAmount
 }
 
-export function formatAmount(amount: number) {
-    return amount.toFixed(2)
+/**
+ * formats a number by removing unnecessary trailing zeros after decimal point
+ * if all decimal places are zeros, returns the whole number
+ * preserves small decimal numbers
+ * @param amount - number or string to format
+ * @returns formatted string representation of the number
+ */
+export const formatAmount = (amount: string | number): string => {
```

Comment:
Issue, blocking: This is not ideal, because a number like 0.011241 will be formatted to "0.011241". We don't need that much precision. The correct formatting would be to format to 2 significant digits, in this case "0.011".